spec/lib/jobs/dell-wsman-add-hotspare-updatexml-spec.js

```diff
@@ -0,0 +1,205 @@
+        smb2Clinet = helper.injector.get('JobUtils.Smb2Client');
+        nfsClinet = helper.injector.get('JobUtils.NfsClient');
+        this.sandbox = sinon.sandbox.create();
+    });
+
+    beforeEach(function() {
+        var options = {
+            driveId: 'Disk.Bay.0:Enclosure.Internal.0-0:RAID.Slot.1-1',
+            volumeId: 'Disk.Virtual.0:RAID.Slot.1-1',
+            hotspareType: 'ghs'
+        };
+        updateXmlJob = new UpdateXmlJob(options, {}, uuid.v4());
+        this.sandbox.stub(configuration, 'get').returns(dellConfigs);
+    });
+
+    afterEach(function() {
+        this.sandbox.restore();
+    });
+
+    describe('_run', function() {
+        it('should handle case: inputed driveId is empty', function() {
+            updateXmlJob.options.driveId = '';
+            return expect(function() {
+                updateXmlJob._run();
+            }).to.throw('The driveId should not be empty.');
+        });
+
+        it('should handle case: share folder is not defined in smlConfig', function() {
+            var dellConfigurations = {
+                services: {}
+            };
+            configuration.get.returns(dellConfigurations);
+            return expect(function() {
+                updateXmlJob._run();
+            }).to.throw('Share folder is not defined in smiConfig.json');
+        });
+
+        it('should run successfully', function() {
+            this.sandbox.stub(UpdateXmlJob.prototype, 'updateXmlForRaid');
+            updateXmlJob._run();
+            return expect(UpdateXmlJob.prototype.updateXmlForRaid).to.be.called;
+        });
+    });
+
+    describe('updateXmlForRaid', function() {
+        beforeEach(function() {
+            this.sandbox.stub(UpdateXmlJob.prototype, 'updateXml');
+        });
+
+        it('should throw error if shareType defined in smiConfig is invalid', function() {
+            var dellConfigurations = {
+                shareFolder: {
+                    shareType: 3
+                }
+            };
+            updateXmlJob.dell = dellConfigurations;
+            this.sandbox.stub(baseJob.prototype, '_done');
+            return updateXmlJob.updateXmlForRaid()
+            .then(function() {
+                expect(baseJob.prototype._done).to.
+                    be.calledWith(new Error('Invalid shareType in smiConfig.json. The shareType should be 0 or 2.'));
+            });
+        });
+
+        it('should update xml in CIFS successfully', function() {
+            var config = {
+                shareFolder: {
+                    shareType: 2,
+                    address: '1.1.1.1',
+                    username: 'username',
+                    password: 'password',
+                }
+            };
+            updateXmlJob.dell = config;
+            this.sandbox.stub(smb2Clinet.prototype, 'readFile').resolves(xmlData);
+            this.sandbox.stub(smb2Clinet.prototype, 'writeFile');
```

Comment:
since `writeFile` is a promise, so also stub it as promise

Reply:
Does it mean that I should let the writeFile stub call resolves(), instead of leaving the stub call its default action, maybe returns() or any else? @yyscamper

Reply:
I got a thought that, in this test case whether the *writeFile* return a promise or return a value, it is not a key point. Just let the UT ensure the promise chain could be fulfilled. Is that right?
Whatever, it could be funny to dig into the default behavior of stub.

Reply:
I mean if the function is a promise, but you not stub it as a promise, your unit-test case may not find the potential problem.

Just like my purpose is to test A, but my test stubs it as B, so I am testing B, even B passes, I still cannot make sure A is OK.